src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -1,11 +1,144 @@
+    ExternalUser,
+    ExternalTeam,
+    RepositoryProjectPathConfig,
+    User,
+)
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+from sentry.ownership.grammar import (
+    parse_code_owners,
+    convert_codeowners_syntax,
+)
+
+from sentry.api.endpoints.project_ownership import ProjectOwnershipSerializer, ProjectOwnershipMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectCodeOwnerSerializer(CamelSnakeModelSerializer):
+    code_mapping_id = serializers.IntegerField(required=True)
+    raw = serializers.CharField(required=True)
+    organization_integration_id = serializers.IntegerField(required=False)
+
+    class Meta:
+        model = ProjectCodeOwners
+        fields = ["raw", "code_mapping_id", "organization_integration_id"]
+
+    def validate(self, attrs):
+        # If it already exists, set default attrs with existing values
+        if self.instance:
+            attrs = {
+                "raw": self.instance.raw,
+                "code_mapping_id": self.instance.repository_project_path_config,
+                **attrs,
+            }
+
+        if not attrs.get("raw", "").strip():
+            return attrs
+        external_association_err = ""
+        # Get list of team/user names from CODEOWNERS file
+        teamnames, usernames, emails = parse_code_owners(attrs["raw"])
+
+        # Check if there exists Sentry users with the emails listed in CODEOWNERS
+        users = User.objects.filter(email__in=emails)
+        users_emails = [user.email for user in users]
+        users_emails_diff = [email for email in emails if email not in users_emails]
+
+        if len(users_emails_diff):
+            external_association_err += f'The following emails do not have an user associated in Sentry: {", ".join(users_emails_diff)}.'
+
+        # Check if the usernames have an association
+        external_users = ExternalUser.objects.filter(external_name__in=usernames)
+
+        external_users_names = [user.external_name for user in external_users]
+        external_users_diff = [name for name in usernames if name not in external_users_names]
+
+        if len(external_users_diff):
+            external_association_err += f'The following usernames do not have an association in Sentry: {", ".join(external_users_diff)}.'
+
+        # Check if the team names have an association
+        external_teams = ExternalTeam.objects.filter(external_name__in=teamnames)
+
+        external_teams_names = [team.external_name for team in external_teams]
+        external_teams_diff = [name for name in teamnames if name not in external_teams_names]
 
+        if len(external_teams_diff):
+            external_association_err += f'The following team names do not have an association in Sentry: {", ".join(external_teams_diff)}.'
 
-class ProjectCodeOwnersEndpoint(ProjectEndpoint):
+        if len(external_association_err):
+            raise serializers.ValidationError({"raw": external_association_err})
```

Comment:
@NisanthanNanthakumar Is this going to look funky if you have multiple error messages joined in on string? Maybe we can provide a list of errors here instead of a string? Not sure the best way of handling this. Alternatively, we could just send only one error message.

Reply:
@scefali hmm, I think it's ok bc its the error for 1 field in the request body and it lists all the problems associated with that 1 field. 
I think sending 1 message at a time would be bad DX. I'd rather know all the problems with my request upfront, rather than piecemeal.

Reply:
@NisanthanNanthakumar ok but maybe we should add new line characters between the messages?